src/MongoDB/Monitoring/TopologyChangedEvent.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2021-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <php.h>
+#include <Zend/zend_interfaces.h>
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "phongo_compat.h"
+#include "php_phongo.h"
+
+zend_class_entry* php_phongo_topologychangedevent_ce;
+
+/* {{{ proto MongoDB\Driver\Monitoring\TopologyChangedEvent TopologyChangedEvent::getNewDescription()
+   Returns this event's new description */
+PHP_METHOD(TopologyChangedEvent, getNewDescription)
+{
+	mongoc_topology_description_t*     topology_description;
+	php_phongo_topologychangedevent_t* intern = Z_TOPOLOGYCHANGEDEVENT_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	topology_description = mongoc_topology_description_new_copy(intern->new_topology_description);
+
+	phongo_topologydescription_init(return_value, topology_description);
+} /* }}} */
+
+/* {{{ proto MongoDB\Driver\Monitoring\TopologyChangedEvent TopologyChangedEvent::getPreviousDescription()
+   Returns this event's previous description */
+PHP_METHOD(TopologyChangedEvent, getPreviousDescription)
+{
+	mongoc_topology_description_t*     topology_description;
+	php_phongo_topologychangedevent_t* intern = Z_TOPOLOGYCHANGEDEVENT_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	topology_description = mongoc_topology_description_new_copy(intern->old_topology_description);
+
+	phongo_topologydescription_init(return_value, topology_description);
+} /* }}} */
+
+/* {{{ proto string TopologyChangedEvent::getTopologyId()
```

Comment:
The [SDAM Monitoring spec](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring-monitoring.rst#topology-ids) doesn't mandate a particular type for topology IDs, but since libmongoc uses ObjectIds I think this method should return an ObjectId as well.

Most functions for initializing an ObjectId are internal to ObjectId.c for the type class itself, but you should find another example of constructing a `MongoDB\BSON\ObjectId` instance in `src/bson.c`, which contains all of the code responsible for converting BSON data to PHP types. The corresponding code therein should serve as a model for what you'll need to do here to prepare the `return_value` zval.